src/main.ts

```diff
@@ -194,6 +199,42 @@ export function createEmitter<T extends Events>(options?: EmitterOptions) {
     }))
   }
 
+  function addGlobalHandler(globalEventHandler: GlobalEventHandler<T>, options?: EmitterOnOptions): () => void {
+    const { signal } = options ?? {}
+
+    if(signal?.aborted) {
+      return () => {}
+    }
+
+    globalHandlers.add(globalEventHandler)
+
+    signal?.addEventListener('abort', () => off(globalEventHandler))
+
+    return () => off(globalEventHandler)
+  }
+
+  function addEventHandler<E extends Event>(event: E, handler: Handler<EventPayload<E>>, options?: EmitterOnOptions): () => void {
+    const { signal } = options ?? {}
+
+    if(signal?.aborted) {
+      return () => {}
+    }
+
+    const existing = handlers.get(event)
+
+    if (existing) {
+      existing.add(handler)
+    } else {
+      handlers.set(event, new Set([handler]))
+    }
```

Comment:
personally I prefer the flow of adding the set if needed but always having the new item be added on it's own line

```suggestion
    if(!handlers.has(event)){
      handlers.set(event, new Set())
    }
    
    handlers.get(event)?.add(handler)
```